tests/integration/test_pip.py

```diff
@@ -89,3 +91,92 @@ def test_pip_packages(
     _ = utils.fetch_deps_and_check_output(
         tmpdir, test_case, test_params, source_folder, test_data_dir, cachi2_image
     )
+
+
+@pytest.mark.parametrize(
+    "test_params,check_cmd",
+    [
+        # Test case checks fetching pip dependencies, generating environment vars file,
+        # building image with all prepared prerequisites and testing if pip packages are present
+        # in built image
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/pip-e2e-test.git",
+                ref="43ff06fdc751ebbbd57f9fe22dc9cc6b5d49a508",
+                packages=(
+                    {
+                        "type": "pip",
+                        "requirements_files": ["requirements.txt"],
+                        "requirements_build_files": [
+                            "requirements-git.txt",
+                            "requirements-https.txt",
+                        ],
```

Comment:
The 3 requirements files are confusing.
* why do we have 3 of them?
* the 2 used as `requirements_build_files` do not contain build dependencies
* `requirements.txt` doesn't look like it will be installable, it's missing transitive dependencies. After pip-compiling, it looks like this:

```requirements.txt
#
# This file is autogenerated by pip-compile with Python 3.9
# by the following command:
#
#    pip-compile --allow-unsafe requirements.in
#
aiowsgi==0.7
    # via -r requirements.in
async-timeout==4.0.2
    # via redis
click==8.1.3
    # via flask
cnr-server @ git+https://github.com/quay/appr.git@58c88e4952e95935c0dd72d4a24b0c44f2249f5b
    # via -r requirements.in
dnspython==2.3.0
    # via python-etcd
flask==2.2.2
    # via
    #   cnr-server
    #   flask-cors
flask-cors==3.0.10
    # via cnr-server
futures==3.0.5
    # via cnr-server
importlib-metadata==6.0.0
    # via flask
itsdangerous==2.1.2
    # via flask
jinja2==3.1.2
    # via flask
markupsafe==2.1.2
    # via
    #   jinja2
    #   werkzeug
peewee==3.15.4
    # via cnr-server
python-etcd==0.4.5
    # via cnr-server
redis==4.4.2
    # via cnr-server
semantic-version==2.10.0
    # via cnr-server
six==1.16.0
    # via flask-cors
urllib3==1.21.1
    # via python-etcd
waitress==2.1.2
    # via aiowsgi
webob==1.8.7
    # via aiowsgi
werkzeug==2.2.2
    # via flask
zipp==3.12.0
    # via importlib-metadata
```